clientlibrary/checkpoint/dynamodb-checkpointer.go

```diff
@@ -195,27 +197,36 @@ func (checkpointer *DynamoCheckpoint) GetLease(shard *par.ShardStatus, newAssign
 
 // CheckpointSequence writes a checkpoint at the designated sequence ID
 func (checkpointer *DynamoCheckpoint) CheckpointSequence(shard *par.ShardStatus) error {
-	leaseTimeout := shard.LeaseTimeout.UTC().Format(time.RFC3339)
-	marshalledCheckpoint := map[string]*dynamodb.AttributeValue{
+	duration := time.Duration(checkpointer.LeaseDuration) * time.Millisecond
+	leaseTimeout := time.Now().UTC().Add(duration)
+	shard.SetLeaseTimeout(leaseTimeout)
+
+	key := map[string]*dynamodb.AttributeValue{
 		LEASE_KEY_KEY: {
 			S: aws.String(shard.ID),
 		},
-		CHECKPOINT_SEQUENCE_NUMBER_KEY: {
-			S: aws.String(shard.Checkpoint),
-		},
-		LEASE_OWNER_KEY: {
-			S: aws.String(shard.AssignedTo),
-		},
-		LEASE_TIMEOUT_KEY: {
-			S: aws.String(leaseTimeout),
-		},
 	}
 
-	if len(shard.ParentShardId) > 0 {
-		marshalledCheckpoint[PARENT_SHARD_ID_KEY] = &dynamodb.AttributeValue{S: &shard.ParentShardId}
+	update := expression.
```

Comment:
Do we need LEASE_OWNER_KEY?

Reply:
No because the current process/worker is the owner. The problem before was that the consumer's lease could have expired and it would set the owner of the shard to `""`. This would cause problems in dynamo at this point in the code since you can't put an empty string to dynamo.
What could happen with this now is that the lease expires and another consumer grabs it but the old owner thinks they still own it. This will cause duplicates. The workaround is to set the failover time (ie the lease duration) high enough to where this doesn't happen. One of the updates I'm planning on doing is a check-and-set operation here where I conditionally update the checkpoint in dynamo to make sure someone else hasn't claimed the shard which will cause an error on the update call.